site/src/pages/WorkspacesPage/WorkspacesPage.test.tsx

```diff
@@ -29,19 +32,52 @@ describe("WorkspacesPage", () => {
     )
 
     // When
-    render(<WorkspacesPage />)
+    renderWithAuth(<WorkspacesPage />)
 
     // Then
     const text = t("emptyCreateWorkspaceMessage", { ns: "workspacesPage" })
     await screen.findByText(text)
   })
 
   it("renders a filled workspaces page", async () => {
-    render(<WorkspacesPage />)
+    renderWithAuth(<WorkspacesPage />)
     await screen.findByText(`${MockWorkspace.name}1`)
     const templateDisplayNames = await screen.findAllByText(
       `${MockWorkspace.template_display_name}`,
     )
     expect(templateDisplayNames).toHaveLength(MockWorkspacesResponse.count)
   })
+
+  it("displays banner for impending deletions", async () => {
+    const MockWorkspaceWithDeletion = {
+      ...MockWorkspace,
+      deleting_at: new Date().toISOString(),
+    }
+
+    const MockWorkspacesResponseWithDeletions = {
+      workspaces: [
+        ...MockWorkspacesResponse.workspaces,
+        MockWorkspaceWithDeletion,
+      ],
+      count: MockWorkspacesResponse.count + 1,
+    }
+
+    jest
+      .spyOn(API, "getEntitlements")
+      .mockResolvedValue(MockEntitlementsWithScheduling)
+
+    jest
+      .spyOn(API, "getWorkspaces")
+      .mockResolvedValue(MockWorkspacesResponseWithDeletions)
+
+    renderWithAuth(<WorkspacesPage />)
+
+    const banner = await screen.findByText(
+      "You have workspaces that will be deleted soon.",
+    )
+    const user = userEvent.setup()
+    await user.click(screen.getByTestId("dismiss-banner-btn"))
+
+    expect(banner).toBeEmptyDOMElement
+  })
```

Comment:
This looks like a rendering test - the test you want to check if something is rendered or not - do you think we should move this to the storybook? I feel this would be better to add it to jest.

Reply:
This test is checking flow - that the banner is visible until the user clicks 'Dismiss' and then it is hidden. However, adding a storybook is a great idea so I'll do that.